### PR TITLE
Textarea: Convert to functional component

### PR DESCRIPTION
**Pull request by westwood846**
See #237.

### CODE REVIEW

File: src/Input/Textarea/Textarea.tsx [hunk 1]
```diff
@@ -9,133 +9,116 @@ import {
...
+
+  return (
+    <TextareaContainer className={classNames('aries-textarea', className)}>
+      <TextareaInput
+        ref={textareaInputRef}
+        placeholder={removeFloatingLabel && label}
+        rows={rows}
+        status={status}
+        disabled={disabled}
+        onBlur={handleFocusChange(onBlur)}
```
Comment (rosslo) [inline src/Input/Textarea/Textarea.tsx]: I think we can refactor `handleFocusChange` to be using it directly.

File: src/Input/Textarea/Textarea.tsx [hunk 2]
```diff
@@ -9,133 +9,116 @@ import {
...
+    removeFloatingLabel,
+    ...restProps
+  } = props;
 
-    return listener;
-  };
+  const [floating, setFloating] = React.useState(false);
+  const [rows, setRows] = React.useState(4);
+  const [minRows] = React.useState(4);
+  const [maxRows] = React.useState(12);
```
Comment (rosslo) [inline src/Input/Textarea/Textarea.tsx]: I think we can hoist `minRows` and `maxRows` out of `Textarea` to be constant.

**Comment by rosslo** [comment]
Btw, I think we need to add `eslint-plugin-react-hooks` to help us checks effect dependencies.

File: src/Input/Textarea/Textarea.tsx (hunk 1 above)
Comment (rosslo) [inline src/Input/Textarea/Textarea.tsx]: Won't it break `useCallback`? Might need to refactor it to be like:
```
  const handleFocusChange = React.useCallback(
    (e: React.FocusEvent<HTMLTextAreaElement>) => {
      setFloating(e.target.value.length > 0);

      if (onBlur !== undefined) {
        return onBlur(e);
      }
    },
    [onBlur]
  );
```
```

File: src/Input/Textarea/Textarea.tsx [hunk 3]
```diff
@@ -145,14 +128,6 @@ interface Props extends React.ComponentPropsWithoutRef<typeof TextareaInput> {
   forwardedRef?: React.RefObject<HTMLTextAreaElement>;
```
Comment (rosslo) [inline src/Input/Textarea/Textarea.tsx]: Sorry, I cannot leave a comment on `onBlur`. Could you help to add `(e: React.FocueEvent<HTMLTextAreaElement>)` for it?

File: src/Input/Textarea/Textarea.tsx [hunk 4]
```diff
@@ -9,133 +9,116 @@ import {
...
+    className,
+    onBlur,
+    onChange,
+    removeFloatingLabel,
+    ...restProps
+  } = props;
 
-    return listener;
-  };
+  const [floating, setFloating] = React.useState(false);
```
Comment (rosslo) [inline src/Input/Textarea/Textarea.tsx]: Should add type check for `useState`, e.g. `React.useState<boolean>`

File: src/Input/Textarea/Textarea.tsx (hunk 2 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: Changed this, thanks :)

File: src/Input/Textarea/Textarea.tsx (hunk 1 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: I added `onBlur` as a dependency for the `useCallback` but I couldn't get your code to work. It complains about type errors and I don't know how to fix that. What exactly is your concern here? It seems to be working the way it is.

File: src/Input/Textarea/Textarea.tsx (hunk 4 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: Oh cool I didn't know about this. Thanks for letting me know. Added this.

**Comment by westwood846** [comment]
@rosslo can you please review the changes?

File: src/Input/Textarea/Textarea.tsx [hunk 5]
```diff
@@ -9,133 +9,117 @@ import {
...
+    onBlur,
+    onChange,
+    removeFloatingLabel,
+    ...restProps
+  } = props;
 
-  handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-    const { minRows, maxRows } = this.state;
+  const [floating, setFloating] = React.useState<boolean>(false);
+  const [rows, setRows] = React.useState<number>(4);
```
Comment (rosslo) [inline src/Input/Textarea/Textarea.tsx]: can use `MIN_ROWS` to instead of `4`

File: src/Input/Textarea/Textarea.tsx (hunk 1 above)
Comment (rosslo) [reply src/Input/Textarea/Textarea.tsx]: Oh, I think you need to do this https://github.com/glints-dev/glints-aries/pull/238#discussion_r384371313 to fix that type error.

Comment (rosslo) [reply src/Input/Textarea/Textarea.tsx]: My concern is `onBlur={handleFocusChange(onBlur)}` is actually recreated fucntion in every render, we can refactor `handleFocusChange` to make it be `onBlur={handleFocusChange}` which should improve performance.

**Comment by westwood846** [comment]
> Btw, I think we need to add `eslint-plugin-react-hooks` to help us checks effect dependencies.

Added an issue at https://github.com/glints-dev/glints-aries/issues/249

File: src/Input/Textarea/Textarea.tsx (hunk 1 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: Updated!

File: src/Input/Textarea/Textarea.tsx (hunk 3 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: Updated!

File: src/Input/Textarea/Textarea.tsx (hunk 5 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: Updated!

Comment (rosslo) [reply src/Input/Textarea/Textarea.tsx]: Oh @westwood846, seems you miss this change. 🙏

**Comment by rosslo** [comment]
> > Btw, I think we need to add `eslint-plugin-react-hooks` to help us checks effect dependencies.
> 
> Added an issue at #249

I added it within this PR. 🙏

File: src/Input/Textarea/Textarea.tsx (hunk 5 above)
Comment (westwood846) [reply src/Input/Textarea/Textarea.tsx]: Whoops :D Fixed.

**Comment by westwood846** [comment]
> > > Btw, I think we need to add `eslint-plugin-react-hooks` to help us checks effect dependencies.
> > 
> > 
> > Added an issue at #249
> 
> I added it within this PR. 🙏

What? Where?

**Comment by rosslo** [comment]
> > > > Btw, I think we need to add `eslint-plugin-react-hooks` to help us checks effect dependencies.
> > > 
> > > 
> > > Added an issue at #249
> > 
> > 
> > I added it within this PR. 🙏
> 
> What? Where?

Lol sorry https://github.com/glints-dev/glints-aries/pull/246

**Pull request merged**